daemon/sdnotify_test.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package daemon
+
+import (
+	"os"
+	"testing"
+)
+
+// TestDaemon
+func TestDaemon(t *testing.T) {
+	// (true, nil) - notification supported, data has been sent
+	// this test should work as long as the test host runs systmed
+	os.Setenv("NOTIFY_SOCKET", "/run/systemd/notify")
+	sent, err := SdNotify("TEST: notification supported, data sent")
+	if !sent || err != nil {
+		t.Errorf("TEST: \"notification supported, data sent\" FAILED")
+	}
+	// (false, nil) - notification not supported
+	os.Unsetenv("NOTIFY_SOCKET")
+	sent, err = SdNotify("TEST: notification not supported")
+	if sent || err != nil {
+		t.Errorf("TEST: \"notification not supported\" FAILED")
+	}
+	// TODO (false, err) - notification supported, but failure happened
```

Comment:
You should be able to reach this case by pointing $NOTIFY_SOCKET to something invalid like a non-existing file, a path that you cannot write to or that is not a socket.